packages/flutter/test/widgets/heroes_test.dart

```diff
@@ -1450,4 +1450,164 @@ void main() {
     ));
     expect(find.text('two'), findsOneWidget);
   });
+
+  testWidgets('Can push/pop on outer Navigator if nested Navigator contains Heroes', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/28042.
+
+    const String heroTag = 'You are my hero!';
+    final GlobalKey<NavigatorState> rootNavigator = GlobalKey();
+    final GlobalKey<NavigatorState> nestedNavigator = GlobalKey();
+    final GlobalKey nestedRouteHeroBottom = GlobalKey();
```

Comment:
Can this key and the next one just be UniqueKey() values or do they really need to be global keys?

Reply:
Changed to UniqueKey here and below.